src/app/core/components/change-display-name/change-display-name.component.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Component,
+  inject,
+  Input,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ButtonModule } from 'primeng/button';
+import { InputTextModule } from 'primeng/inputtext';
+import { FormControl, ReactiveFormsModule, Validators } from '@angular/forms';
+import { AuthenticationService } from '../../services/authentication.service';
+
+@Component({
+  selector: 'app-change-display-name',
+  standalone: true,
+  imports: [CommonModule, ButtonModule, InputTextModule, ReactiveFormsModule],
+  templateUrl: './change-display-name.component.html',
+  styleUrls: ['./change-display-name.component.scss'],
+})
+export class ChangeDisplayNameComponent implements OnChanges {
+  authenticationService = inject(AuthenticationService);
+
+  @Input() displayName?: string;
+
+  displayNameFormControl = new FormControl(
+    { value: '', disabled: true },
+    {
+      validators: [Validators.required, Validators.maxLength(12)],
+    },
+  );
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (this.displayName)
+      this.displayNameFormControl.setValue(this.displayName);
+  }
+
+  toggleDisplayNameDisabled() {
```

Comment:
you repeat `display name` too much. The component itself sets the namespace. You can just use a control or form control in the component itself. No need to repeat the names. Makes it too noisy and less readable.